### DISCUSSION TITLE
Can't connect to ProFTP using Fetch on Mac OS 7, receive Illegal PORT command

### DISCUSSION THREAD

**Discussion by aptonline** [retronas/discussions#47] [Q&A]
I'm using Fetch (version 4.0.3) to attempt to connect to the ProFTP install on RetroNAS but receive the following error:

![Picture 2](https://user-images.githubusercontent.com/190136/183086710-5460e28d-63b8-4916-867d-e82be882e587.jpg)

and in the transcript:
```
PWD
257 "/" is the current directory
PORT 10,0,2,15,114,235
500 Illegal PORT command
ftp_list: -30000 (state == GETTING_LIST)
```
I can connect successful to the FTP server using Mac OS Mojave etc but so I know the server is working correctly.

Any ideas why it's not working?

**Comment by sairuk** [comment]
Converted to discussion, ftp as deployed by retronas is functional with other tools.


What mode does fetch use? `Illegal PORT Command` is a ProFTPD specific error when the requested ip address isn't accessible by the ftpd in this case `10.0.2.15` is the ip. Is your RetroNAS split net? what ip is the ftpd listening on?

can check with something like 
`sudo ss -lnp | grep proftpd`

**Reply by sairuk** [reply]
Looks like fetch can use PASV mode (not sure if its in your version) which was going to be my next suggestion (assuming this is the correct software)

https://fetchsoftworks.com/fetch/help/Contents/FAQ/TroubleshootingFAQ.html#InvalidPORTCommand

https://fetchsoftworks.com/fetch/help/Contents/Concepts/ActiveAndPassive.html

**Reply by sairuk** [reply]
According to the 4.0.3 doco it does support PASV under Firewall Preferences

https://fetchsoftworks.com/fetch/download/Fetch4Help.pdf?direct=1

**Reply by aptonline** [reply]
Thanks for your reply. here is the output from the command:

`tcp   LISTEN 0      128                                                                                                      *:21                     *:*      users:(("proftpd",pid=13003,fd=0))  `  

I've discovered that if I set the initial directory path to `/home/me` (which is the default account) I can get a listing of that directory but if I go to the root (`/`) or the path to where the storage is setup (`/ds918`) I receive the Illegal Port command again.

I am able to successfully navigate any directory using macOS Mojave or other devices using the same credentials.

Tried with PASV mode on and off and the same issue.
